ci/deps/pyspark.yml

```diff
@@ -1 +1 @@
-pyspark>=2.4.3
```

Comment:
wait, i thought we are going to run *both* spark2 and spark3 ?

Reply:
whether 3.7 on spark2 and 3.8 spark3 is fine

Reply:
Yep this is correct (the PySpark 2 dependency is defined in https://github.com/ibis-project/ibis/blob/master/ci/deps/pyspark-min.yml)

Here's a breakdown:

- `Main / Tests PySpark / Spark (3.7)` CI Job:
  - Uses `ci/deps/pyspark-min.yml`
  - PySpark 2
  - Python 3.7

- `Main / Tests PySpark / Spark (3.8)` CI Job:
  - Uses `ci/deps/pyspark.yml`
  - PySpark 3
  - Python 3.8

Reply:
Where are those specified? I don't see it in this PR.

Reply:
The CI changes were broken up into a couple of PRs:

**#2359 CI: Splitting OmniSci and Spark jobs, and allowing to install minimum supported dependencies** (merged) :
- Split OmniSci/PySpark/Spark job
- Create `xxxx-min.yml` files
- (In this PR, only running tests on PySpark 2 + Python 3.7)

**#2363 CI: Enable UDF tests for PySpark and Spark backends** (merged):
- Enable UDF tests for the PySpark/Spark job
- Fix dependencies in `pyspark-min.yml` and `spark-min.yml` so that those UDF tests pass

**This PR**:
- Enable the `Main / Tests PySpark / Spark (3.8)` job
- Fix tests so that all tests pass in both the `Main / Tests PySpark / Spark (3.7)` job and the `Main / Tests PySpark / Spark (3.8)` job